### PR TITLE
Review sampling bias

### PR DESCRIPTION
**Pull request by wlangera**
@sangovae and @sunray1 I will add the code of @Mmyahaya and me and my review for #48 in this draft PR.
So that you have an idea what I am doing. I will let you know if it is final so you can again review.

### CODE REVIEW

**Comment by wlangera** [comment]
Changes **sample_observations.R**:

- update documentation
   - make conform with other functions of the package
   - remove mentioning SpatRaster object, I will add an issue for this
   - add example (+ visualise)
- add `samping_bias_manual()` function

Changes **apply_polygon_sample_bias.R**:

- update documentation
   - make conform with other functions of the package
   - add example (+ visualise)
-  fix bug matrix to vector

Changes **samping_bias_manual.R**:

- update documentation
   - make conform with other functions of the package
   - add example (+ visualise)
- update checks
   - for every input argument a check
   - check for numeric column bias_weight
   - add check such that numbers larger than 1 are integers

Changes in unit tests:

I just made sure all tests worked. But we need to make more. But lets do that another time ...

**Comment by sunray1** [comment]
@wlangera let me know when you are finished making adjustments and I will review

**Comment by wlangera** [comment]
@sunray1 I am finished :) it is no longer a draft

**Review by sangovae — APPROVED** [review]
I added some more commits to fix little things, not sure if these are also taken in this pull request then?

**Comment by sunray1** [comment]
Yes, they have - reviewing now.

File: R/sample_observations.R [hunk 1]
```diff
@@ -60,21 +63,58 @@
...
+#' sample_observations(
+#'   occurrences_sf,
+#'   detection_probability = 0.8,
+#'   sampling_bias = "polygon",
+#'   bias_area = bias_area,
+#'   bias_strength = 2,
+#'   seed = 123
+#'   )
+#'
+#' # Sample observations with sampling bias in a polygon
```
Comment (sunray1) [inline R/sample_observations.R:92]:
```suggestion
#' # Sample observations with sampling bias given manually in a grid
```

File: R/sample_observations.R [hunk 2]
```diff
@@ -17,36 +17,39 @@
...
+#' applied to the sampling of occurrences. Higher weights mean a higher
+#' probability of sampling. Weights can be numeric values between 0 and 1 or
+#' positive integers that will be rescaled to values between 0 and 1.
 #' @param seed A positive numeric value. The seed for random number generation
 #' to make results reproducible. If `NA` (the default), no seed is used.
 #'
 #' @returns An sf object with POINT geometry containing the locations of the
 #' sampled observations, a `detection_probability` column containing the
-#' detection probability for each observation (will be the same for all), a
+#' detection probability for each observation (will be the same for all), and a
```
Comment (sunray1) [inline R/sample_observations.R:34]:
```suggestion
#' detection probability for each observation (will be the same for all), a
```

File: R/sample_observations.R [hunk 3]
```diff
@@ -17,36 +17,39 @@
...
 #' `bias_weight` column containing the sampling probability based on sampling
 #' bias, a `sampling_probability` column containing the combined sampling
```
Comment (sunray1) [inline R/sample_observations.R:36]:
```suggestion
#' bias, and a `sampling_probability` column containing the combined sampling
```

**Review by sunray1 — CHANGES_REQUESTED** [review]
Didn't review the manual files - I imagine they may need to be re-built though.

File: R/sampling_bias_manual.R
[the remark this answers is not in the document]
Comment (wlangera) [reply R/sampling_bias_manual.R:1]: yes, but lets leave that for now. Can you make an issue for that?

**Comment by wlangera** [comment]
yes @sunray1 I will pul again from main and rebuild, then I will merge

File: R/sampling_bias_manual.R
[the remark this answers is not in the document]
Comment (sunray1) [reply R/sampling_bias_manual.R:1]: See #55

**Review by sunray1 — APPROVED** [review]

**Pull request merged**